semgrep-core/parsing/Parse_rust_tree_sitter.ml

```diff
@@ -2143,22 +2125,41 @@ and map_statement (env : env) (x : CST.statement): G.stmt list =
    | `Item x -> map_item env x
   )
 
-and map_trait_block (env : env) ((v1, v2, v3) : CST.trait_block): G.stmt list =
+and map_trait_block (env : env) ((v1, v2, v3) : CST.trait_block): G.field list G.bracket =
   let lbrace = token env v1 (* "{" *) in
-  let stmts = List.map (map_trait_block_item env) v2 in
+  let fields = List.map (map_trait_block_item env) v2 in
   let rbrace = token env v3 (* "}" *) in
-  stmts
+  (lbrace, fields, rbrace)
 
-and map_trait_block_item (env : env) ((v1, v2) : CST.trait_block_item): G.stmt =
+and map_trait_block_item (env : env) ((v1, v2) : CST.trait_block_item): G.field =
   let outer_attrs = List.map (map_outer_attribute_item env) v1 in
   match v2 with
-  | `Const_item x -> map_const_item env x
+  | `Const_item x -> G.FieldStmt (map_const_item env x)
   | `Func_sign_with_defa_item x ->
       let def = map_function_signature_with_default_item env x in
-      G.DefStmt def |> G.s
-  | `Asso_type x -> map_associated_type env x
+      G.FieldStmt (G.DefStmt def |> G.s)
+  | `Asso_type x -> G.FieldStmt (map_associated_type env x)
   | `Macro_invo x -> let invo = map_macro_invocation env x in
-      G.ExprStmt (invo, sc) |> G.s
+      G.FieldStmt (G.ExprStmt (invo, sc) |> G.s)
+
+and map_higher_ranked_trait_bound (env : env) ((v1, v2, v3) : CST.higher_ranked_trait_bound): (G.type_parameter list * G.type_) =
+  let for_ = token env v1 (* "for" *) in
```

Comment:
BTW at some point you may want to remove the [@@@ warning "-26-27"] directives at the top
to get the unused_variable check from the OCaml compiler back on, so you'll have to prefix
those unused tokens with an '_', e.g. let _for = token ... here.